go-controller/pkg/clustermanager/subnetallocator/named_subnet_allocator.go

```diff
@@ -0,0 +1,191 @@
+package subnetallocator
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+)
+
+type recordSubnetUsageFn func(v4SubnetsAllocated, v6SubnetsAllocated float64)
+type recordSubnetCountFn func(v4SubnetsAllocated, v6SubnetsAllocated float64)
+
+type namedSubnetAllocator struct {
```

Comment:
hmm I get why you had to split the common functions out into a new allocator struct to reuse this for both zone and host subnet allocation but `named`SubjectAllocator seems to not tell anything useful. But we seem to have used up `baseSA` and `SA already` and this is already 3rd layer of building embedding the other two..
let me see if someone has some ideas for a better "name" :D